docs/contribute/setup.md

```diff
@@ -0,0 +1,150 @@
+
+## Clone the repository
+
+First off you'll need your own copy of the `pydata-sphinx-theme` codebase.
+You can clone it for local development like so:
+
+1. **Fork the repository** so you have your own copy on GitHub.
+   See [the GitHub forking guide](https://docs.github.com/en/get-started/quickstart/fork-a-repo) for more information.
+2. **Clone the repository locally** so that you have a local copy to work from:
+
+   ```console
+   $ git clone https://github.com/{{ YOUR USERNAME }}/pydata-sphinx-theme
+   $ cd pydata-sphinx-theme
+   ```
+
+## Install your tools
+
+Building a Sphinx site uses a combination of Python and Jinja to manage HTML, SCSS, and Javascript.
+To simplify this process, we use a few helper tools:
+
+- [The Sphinx Theme Builder](https://sphinx-theme-builder.readthedocs.io/en/latest/) to automatically perform compilation of web assets.
+- [pre-commit](https://pre-commit.com/) for automatically enforcing code standards and quality checks before commits.
+- [nox](https://nox.thea.codes/), for automating common development tasks.
+
+In particular, `nox` can be used to automatically create isolated local development environments with all of the correct packages installed to work on the theme.
+The rest of this guide focuses on using `nox` to start with a basic environment.
+
+```{seealso}
+The information on this page covers the basics to get you started, for information about manually compiling assets, see [](manual.md).
+```
+
+### Setup `nox`
+
+To start, install `nox`:
+
+```console
+$ pip install nox
+```
+
+You can call `nox` from the command line in order to perform common actions that are needed in building the theme.
+`nox` operates with isolated environments, so each action has its own packages installed in a local director (`.nox`).
+For common development actions, you'll simply need to use `nox` and won't need to set up any other packages.
+
+### Setup `pre-commit`
+
+`pre-commit` allows us to run several checks on the codebase every time a new Git commit is made.
+This ensures standards and basic quality control for our code.
+
+Install `pre-commit` with the following command:
+
+```console
+$ pip install pre-commit
+```
+
+then navigate to this repository's folder and activate it like so:
+
+```console
+$ pre-commit install
+```
+
+This will install the necessary dependencies to run `pre-commit` every time you make a commit with Git.
+
+## Build the documentation
+
+Now that you have `nox` installed and cloned the repository, you should be able to build the documentation locally.
+
+To build the documentation with `nox`, run the following command:
+
+```console
+$ nox -s docs
+```
+
+This will install the necessary dependencies and build the documentation located in the `docs/` folder.
+They will be placed in a `docs/_build/html` folder.
+If the docs have already been built, it will only build new pages that have been updated.
+You can open one of the HTML files there to preview the documentation locally.
```

Comment:
Do we want/need to mention `http.server` to serve that content easily and navigate into the pages?

Reply:
that's a cool idea - I can't remember the command, wanna add it?

Reply:
https://github.com/choldgraf/pydata-sphinx-theme/pull/4